### ISSUE TITLE
Toolchain 2019-02-25 does not recognize Tensorflow and @differentiable

### ISSUE THREAD

**Issue by Theo58** [tensorflow/swift#125]
We are using [JIRA](https://bugs.swift.org) instead of GitHub issues to track
our work in order to integrate more effectively with the rest of the Swift
ecosystem.

Before opening a new issue, please [search the existing
issues](https://bugs.swift.org/issues/?jql=project%20%3D%20TF) to see if your
issue has already been asked. If it's open, you can _Add Vote_ to request
the issue be prioritized. And you can start watching the issue to receive
updates.

If there are no related issues in JIRA, please file a new ticket by using one
of the links below corresponding to the nature of your issue:

  - [Create a bug](https://bugs.swift.org/secure/CreateIssue.jspa?issuetype=10006&pid=10100)
  - [Create a new feature request (Story)](https://bugs.swift.org/secure/CreateIssue.jspa?issuetype=10001&pid=10100)
  - [Create a new clean up / internal improvement](https://bugs.swift.org/secure/CreateIssue.jspa?issuetype=10003&pid=10100)

**Comment by Theo58** [comment]
Moved the issue to Jira

**Comment by Theo58** [comment]
The new 0.2 release of the toolchain has the same issue. It does not recognize TensorFlow and producers other errors

**Comment by rxwei** [comment]
Is your development environment Xcode?

If you are using Xcode, it is a known issue of Xcode, and it's documented in the [Install Swift for TensorFlow](https://github.com/tensorflow/swift/blob/master/Installation.md#installation) manual where you downloaded the toolchain. Follow the installation steps and you should be able to use it.

**Comment by Theo58** [comment]
I do use xCode 102. beta. I did the install instructions as described.
 1:  set the build system to "legacy build system"
 2: add libtensorflow.so to the linked frameworks and libraries.
 3: selected the option -O (speed optimization)
with toolchain 2019-02-20 it works (although I still have strange issues like it does not recognize the statement Context)
With toolchain 2019-02-25 and release 0.2 it does not work. I do make sure that I load the libtensorflow.so out of the correct toolchain.

**Comment by rxwei** [comment]
I'm sorry that I am unable to reproduce this in my Xcode 10.2 beta. Setting the build system to "Legacy Build System" is key. I'm not sure why that didn't resolve your issue.

Could you try removing other toolchains from your system and just try out the Release 0.2 version?

**Comment by Theo58** [comment]
Hallo. Yes, this did help. I first selected the standard toolchain and removed the libtensorflow.so. Then I removed all older toolchains out of xcode (there is a button for it). then I selected the new toolchain and added again the libtensorflow.so. Now it works. Ik recognizes TensorFlow as a library and now generates all kind of error that have to do with changes in the release (like changing @differentiable(wrt: (self, input)) into @differentiable). So I will figure out how to fix this. But this issue is now solved. Thanks for the suggestion. remaining question. I browsed all the GitHub parts and also look a lot at the TensorFlow interface definition (jump to definition in xcode) to learn about all kind of stuff but it is difficult. Like why Cov2D requires a 4D tensor as input and not a 2D (which I understand for the second time you use it, but not for the first layer which is looking at a 2D image if you use a monochrome picture). What is the best place to look for these kind of things?

Thanks for the suggestion clearing the old stuff. I will close the issue.
